components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -7244,4 +7255,30 @@ private String getPrimaryUserStoreDomain() {
         }
         return primaryUSDomain;
     }
+
+    /**
+     * Checks whether the given set of claim URIs contains any non-account state claims.
+     * Any non-account state claim is considered as a real profile attribute.
+     *
+     * @param claimUris Set of claim URIs to check.
+     * @return true if there is any non-account state claim, false otherwise.
+     */
+    private boolean hasAnyNonAccountStateClaims(Set<String> claimUris) {
```

Comment:
This method name should reflect the top comment I added. I think it should be something like includesOnlyAccountManagementFlowInitClaims